src/jit/morph.cpp

```diff
@@ -10638,7 +10638,12 @@ GenTree* Compiler::fgMorphRecognizeBoxNullable(GenTree* compare)
     }
 
     // replace the box with an access of the nullable 'hasValue' field which is at the zero offset
-    GenTree* newOp = gtNewOperNode(GT_IND, TYP_BOOL, opCall->gtCall.gtCallArgs->gtOp.gtOp2->gtOp.gtOp1);
+    GenTreePtr arg = opCall->gtCall.gtCallArgs->gtOp.gtOp2->gtOp.gtOp1;
```

Comment:
I would add a comment here stating the we check for the following three cases and early out, because it is unsafe to optimize.